include/fmt/format.h

```diff
@@ -3669,16 +3843,16 @@ operator"" _a(const wchar_t *s, std::size_t) { return {s}; }
 #endif // FMT_USE_USER_DEFINED_LITERALS
 FMT_END_NAMESPACE
 
-#define FMT_STRING(s) [] { \
-    typedef typename std::remove_cv<std::remove_pointer< \
-      typename std::decay<decltype(s)>::type>::type>::type ct; \
-    struct str : fmt::compile_string { \
-      typedef ct char_type; \
-      FMT_CONSTEXPR operator fmt::basic_string_view<ct>() const { \
-        return {s, sizeof(s) / sizeof(ct) - 1}; \
-      } \
-    }; \
-    return str{}; \
+#define FMT_STRING(s)                                                     \
+  [] {                                                                    \
+    struct str : fmt::compile_string {                                    \
+      typedef typename std::remove_cv<std::remove_pointer<                \
+          typename std::decay<decltype(s)>::type>::type>::type char_type; \
+      FMT_CONSTEXPR operator fmt::basic_string_view<char_type>() const {  \
+        return {s, sizeof(s) / sizeof(char_type) - 1};                    \
+      }                                                                   \
+    };                                                                    \
+    return str{};                                                         \
```

Comment:
What is this change for?

Reply:
I moved the `char_type` typedef into the `str` struct, because it's useful for `prepare()` while handling compile-time format strings. Then, while formatting the pull request's code, clang-format detected that the define has changes, so it formatted whole define. That's why there is so many changes. The actual code change is renaming `ct` => `char_type` and moving it into the `str`.

I know that because of reformatting it mostly looks like whitespace change but can we keep it like this? It'll be pain to track manual and clang-format formatting while working on this pr and with this define. The problem will be resolved with #988

Reply:
> I know that because of reformatting it mostly looks like whitespace change but can we keep it like this?

Sure.